.github/workflows/build_and_test.yaml

```diff
@@ -20,31 +20,25 @@ jobs:
   build-and-test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - name: Check out code
+        uses: actions/checkout@v3
       - name: Run yamllint
         run: |
-          MAKE_IN_DOCKER=1 make lint-yamllint
-      - name: Run Codespell
-        uses: codespell-project/actions-codespell@master
-        with:
-          # can't reuse tools/codespell/.codespell.skip, shell is not supported
-          skip: .git,*.png,*.woff,*.woff2,*.eot,*.ttf,*.jpg,*.ico,*.svg,go.mod,go.sum
-          ignore_words_file: 'tools/codespell/.codespell.ignorewords'
-          check_filenames: true
-          check_hidden: true
-      - uses: actions/setup-go@v3
+          MAKE_IN_DOCKER=1 make lint.yamllint
+      - name: Run codespell
+        run: |
+          MAKE_IN_DOCKER=1 make lint.codespell
+      - name: Run golangci-lint 
+        run: |
+          MAKE_IN_DOCKER=1 make lint.golint
+      - name: Set up Go
+        uses: actions/setup-go@v3
         with:
           go-version: ${{ env.GO_VERSION }}
-          cache: true
-      - name: Run golangci-lint
-        uses: golangci/golangci-lint-action@v2
-        with:
-          version: v1.46.2
-          args: --build-tags=e2e
-      - name: Build
-        run: make build-all
+      - name: Build Binaries
+        run: MAKE_IN_DOCKER=1 make build.multiarch
```

Comment:
it was intentional to build the binary on the host, please revert this

Reply:
I do not know why, I tried to use make build-all to build it on the host, but see it failed, it looks like related to the lint steps.

Reply:
anything show up in the raw CI logs ?

Reply:
```
2022-06-30T06:42:46.3746259Z mkdir: cannot create directory ‘/home/runner/work/gateway/gateway/bin/darwin’: Permission denied
```
you probably dont want to create any dir, reverting
```
# Set Output Directory Path
 ifeq ($(origin OUTPUT_DIR),undefined)
 OUTPUT_DIR := $(ROOT_DIR)/bin
 $(shell mkdir -p $(OUTPUT_DIR))
 endif

 # Set TMP Directory Path
 ifeq ($(origin TMP_DIR),undefined)
 TMP_DIR := $(OUTPUT_DIR)/tmp
 $(shell mkdir -p $(TMP_DIR))
 endif
 ```
 and letting `go build` create the binary in the right loc should do the trick

Reply:
Thanks @arkodg It seems like CI passed.